src/main/java/org/openshift/jenkins/plugins/openshiftlogin/OpenShiftOAuth2SecurityRealm.java

```diff
@@ -257,11 +258,13 @@ public void initialize(HttpRequest request) throws IOException {
                                 }
                             });
                     GenericUrl url = new GenericUrl(serverPrefix + "/oapi/v1/users/~");
-
+                    
+                    // do self-sar here to see if you have the "admin" or "view" verb in the "build.openshift.io" api group on the "jenkins" resource 
```

Comment:
OK progress ... after searching through doc/code, I found the `Scopes` fields in three token types in `pkg/oauth/api/types.go`.  I found an `Add` method in `pkg/oauth/scope/scope.go`, which is used by the `GrantHandler` interface.  I've only found HTTP POST level access to the `GrantHandler`, not a direct `oc` or `oadm` entryway into adding scopes (unlike roles).

So I ran an `oc get` for the 3 oauth types that had scope fields ... two get hits:
```
gmontero ~ $ oc get oauthaccesstoken -o json
{
    "kind": "List",
    "apiVersion": "v1",
    "metadata": {},
    "items": [
        {
            "kind": "OAuthAccessToken",
            "apiVersion": "v1",
            "metadata": {
                "name": "txcQaEHfabHd2N4-zNEcdALalTkoHSDHRCxXKI05Fzo",
                "selfLink": "/oapi/v1/oauthaccesstokens/txcQaEHfabHd2N4-zNEcdALalTkoHSDHRCxXKI05Fzo",
                "uid": "6635559f-6556-11e6-a4ef-68f7287398a7",
                "resourceVersion": "499",
                "creationTimestamp": "2016-08-18T15:14:03Z"
            },
            "clientName": "openshift-challenging-client",
            "expiresIn": 86400,
            "scopes": [
                "user:full"
            ],
            "redirectURI": "https://192.168.0.9:8443/oauth/token/implicit",
            "userName": "test",
            "userUID": "6634711e-6556-11e6-a4ef-68f7287398a7"
        }
    ]
}
gmontero ~ $ oc get oauthclientauthorization -o json
{
    "kind": "List",
    "apiVersion": "v1",
    "metadata": {},
    "items": [
        {
            "kind": "OAuthClientAuthorization",
            "apiVersion": "v1",
            "metadata": {
                "name": "test:system:serviceaccount:test:default",
                "selfLink": "/oapi/v1/oauthclientauthorizations/test:system:serviceaccount:test:default",
                "uid": "53d50d3a-655f-11e6-a4ef-68f7287398a7",
                "resourceVersion": "1382",
                "creationTimestamp": "2016-08-18T16:17:58Z"
            },
            "clientName": "system:serviceaccount:test:default",
            "userName": "test",
            "userUID": "6634711e-6556-11e6-a4ef-68f7287398a7",
            "scopes": [
                "user:info"
            ]
        }
    ]
}
```
One is `user:full` and just seems to reference the `test` id, the other seems to be for the default serviceaccount but also references the `test` id and is `user:info`

I re-ran my test, but the token codes returned don't match the token for the default service account.  Should they match up?

I'll circle back and see about updating the client auth for the default sa I found with user:info, but wanted to at least confirm if I'm on the right path and if you can say from what I provided that things are working correctly?

thanks